search_params.py

```diff
@@ -0,0 +1,132 @@
+import argparse
+import logging
+import os
+import time
+import yaml
+from datetime import datetime
+from pathlib import Path
+
+from ray import tune
+
+from libmultilabel import data_utils
+from libmultilabel.evaluate import evaluate
+from libmultilabel.model import Model
+from libmultilabel.utils import ArgDict, set_seed, init_device
+
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s %(levelname)s:%(message)s')
+
+
+def training_function(config):
+    model_config = ArgDict(config)
+    datasets = data_utils.load_datasets(model_config)
+    word_dict = data_utils.load_or_build_text_dict(model_config, datasets['train'])
+    classes = data_utils.load_or_build_label(model_config, datasets)
+
+    model = Model(model_config, word_dict, classes)
+    model.train(datasets['train'], datasets['val'])
+    model.load_best()
+
+    # return best eval metric
+    dev_loader = data_utils.get_dataset_loader(model_config, datasets['val'], model.word_dict, model.classes, train=False)
+    results = evaluate(model_config, model, dev_loader, split='val', dump=True)
+    yield results
```

Comment:
we use yield instead of return because this fun is called/resumed several times?